File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
def solution(n, k):
    answer = 0
    word = ''
    while(n):
        word = str(n%k) + word
        n //=k
        
    def isSosu(x):
        if x<2: return False
        if x == 2: return True
        for i in range(2,int(x**(0.5))+1):
            if x%i == 0:
                return False
        return True
    
    lst = word.split('0')

    for l in lst:
        if l == '':
            continue
        num = int(l)
        if isSosu(num):
            answer+=1
    
    return answer

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.js
function solution(n, k) {
    var answer = 0;
    var str = ''

    while(n>0){
        str = (n%k).toString() + str
        n = Math.floor(n/k)
    }
    
    let values = str.split('0')
    
    function isPrime(num){
        if(num < 2) return false;
        for (let i =2;i<Math.floor(Math.sqrt(num))+1;i++){
            if(num%i === 0){
                return false
            }
        }
        return true
    }
    for(let value of values) {
        if (value === '') continue
        
        value = Number(value)
        if (isPrime(value)) {
            answer += 1
        }
        
    }
    
    return answer;
}